exams/pearson/audit.py

```diff
@@ -0,0 +1,272 @@
+"""Exam auditing"""
+import base64
+import logging
+import os
+import re
+
+from boto.s3 import (
+    connection,
+    key,
+)
+from django.conf import settings
+from django.core.exceptions import ImproperlyConfigured
+import gnupg
+
+SSE_ENCRYPTION_ALGORITHM = 'AES256'
+REQUIRED_SETTINGS = [
+    'EXAMS_AUDIT_ENCRYPTION_FINGERPRINT',
+    'EXAMS_AUDIT_ENCRYPTION_PUBLIC_KEY',
+]
+
+log = logging.getLogger(__name__)
+
+
+class S3AuditStorage:
+    """Audit storage mechanism for S3"""
+
+    def get_connection(self):
+        """
+        Creates a connection to S3
+
+        Raises:
+            ImproperlyConfigured
+                if a required setting is not set
+        Returns:
+            boto.s3.connection.S3Connection
+                a connection to S3
+        """
+        if not all((
+                settings.EXAMS_AUDIT_AWS_ACCESS_KEY_ID,
+                settings.EXAMS_AUDIT_AWS_SECRET_ACCESS_KEY,
+        )):
+            raise ImproperlyConfigured(
+                "One of EXAMS_AUDIT_AWS_ACCESS_KEY_ID or EXAMS_AUDIT_AWS_SECRET_ACCESS_KEY is required but not set"
+            )
+        return connection.S3Connection(
+            settings.EXAMS_AUDIT_AWS_ACCESS_KEY_ID,
+            settings.EXAMS_AUDIT_AWS_SECRET_ACCESS_KEY
+        )
+
+    def get_bucket(self):
+        """
+        Gets the configured bucket
+
+        Raises:
+            ImproperlyConfigured
+                if a required setting is not set
+        Returns:
+            boto.s3.bucket.Bucket
+                the S3 bucket for storage
+        """
+        if settings.EXAMS_AUDIT_S3_BUCKET is None:
+            raise ImproperlyConfigured('Setting EXAMS_AUDIT_S3_BUCKET is required but not set')
+        return self.get_connection().get_bucket(settings.EXAMS_AUDIT_S3_BUCKET)
+
+    def get_s3_key(self, filename, file_type):
+        """
+        Determines the S3 key to store the file under
+
+        Args:
+            filename (str):
+                filename of the encrypted file
+            file_type (str):
+                type of the encrypted file
+
+        Returns:
+            boto.s3.key.Key:
+                the key to store the file in
```

Comment:
I've seen both styles (and looking in search/indexing_api.py I see some stlye mixing, which is even worse), perhaps we can collectively decide on which to standardize?